### PR TITLE
Introduce UserService to centralize logic.

### PR DESCRIPTION
**Pull request by demiankatz**
This PR begins to introduce the Db\Service infrastructure that will be a key component of our eventual Doctrine migration (see #2233). As of this PR, only one service (UserService) is introduced, and it is currently used for the sole purpose of centralizing encryption logic related to catalog passwords. Some of the classes and infrastructure introduced here may seem excessive or pointless at this stage, but they will serve a greater purpose as the code evolves further. The intent is to keep the dev branch as parallel with the work in #2233 as possible to minimize unnecessary conflicts during ongoing development.

TODO
- [x] Fix broken tests

### CODE REVIEW

File: module/VuFind/src/VuFind/Db/Service/UserService.php [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+     * logic is all in one location
+     *
+     * @param string $text    The text to be encrypted or decrypted
+     * @param bool   $encrypt True if we wish to encrypt text, False if we wish to
+     * decrypt text.
+     *
+     * @return string|bool    The encrypted/decrypted string
+     * @throws \VuFind\Exception\PasswordSecurity
+     */
+    public function encryptOrDecrypt($text, $encrypt = true)
```
Comment (EreMaijala) [inline module/VuFind/src/VuFind/Db/Service/UserService.php]: Could we expose something like `encryptData` and `decryptData` and make this protected? That would make the intent of the method calls immediately obvious.

**Review by EreMaijala — CHANGES_REQUESTED** [review]
I think this is a good start. 

How would you feel about prefixing "service" with "db" in most places? Like
AbstractDbService
DbServiceAwareTrait
DbServiceAwareInterface
protected $dbServiceManager;

I think that would alleviate some potential confusion and possible clashes with any other services.

**Comment by demiankatz** [comment]
@EreMaijala, your suggestions all make sense to me. It will take a little time to apply them first to the doctrine branch and then backport them to here, but I think it's worth the effort. I'll request a new review once I've had time to work through it all.

**Comment by demiankatz** [comment]
Thanks again, @EreMaijala -- I've made your suggested changes here and in #2233.

**Review by EreMaijala — APPROVED** [review]

**Pull request merged**